pinot-common/src/main/java/org/apache/pinot/common/metadata/RowMetadata.java

```diff
@@ -23,16 +23,19 @@
 
 
 /**
- * A class that provides relevant row-level metadata for rows ingested into a segment.
+ * A class that provides relevant row-level metadata for rows indexed into a segment.
  *
- * Currently this is relevant for rows ingested into a mutable segment.
+ * Currently this is relevant for rows ingested into a mutable segment - the metadata is expected to be
+ * provided by the underlying stream.
  */
 @InterfaceAudience.Public
 @InterfaceStability.Evolving
 public interface RowMetadata {
 
   /**
-   * Return the ingestion timestamp of the row.
+   * Return the timestamp associated with when the row was ingested upstream.
```

Comment:
Please also include in the comments, what value an implementation should return if it does not have the information. Not all streams may provide a timestamp when an event was ingested upstream.